container-search/src/main/java/com/yahoo/search/query/properties/QueryProperties.java

```diff
@@ -182,9 +181,11 @@ public Object get(CompoundName key,
         if (key.first().equals(Ranking.RANKING)) {
             Ranking ranking = query.getRanking();
             if (key.size() > 2) {
+                String rest = key.rest().rest().toString();
                 // pass the portion after "ranking.features/properties" down
-                if (key.get(1).equals(Ranking.FEATURES)) return ranking.getFeatures().getObject(key.rest().rest().toString());
-                if (key.get(1).equals(Ranking.PROPERTIES)) return ranking.getProperties().get(key.rest().rest().toString());
+                if (key.get(1).equals(Ranking.FEATURES)) return ranking.getFeatures().getObject(rest);
+                if (key.get(1).equals(Ranking.PROPERTIES)) return ranking.getProperties().get(rest);
+                if (key.get(1).equals(Matching.ELEMENT_GAP)) return ranking.getMatching().getElementGapForField(rest);
```

Comment:
I ideally want the property name hierarchy match the object hierarchy (property "a.b.c" implies we have Query.getA().getB().getC()), so maybe we can add a ".matching" to the name or putt it up to ranking?